### PR TITLE
Account fields

### PR DESCRIPTION
**Pull request by CityVibes**
Support for account creation with new account info.

### CODE REVIEW

**Review by idokleinman — APPROVED** [review]

File: .idea/gradle.xml [hunk 1]
```diff
@@ -14,13 +14,7 @@
             <option value="$PROJECT_DIR$/example_app" />
           </set>
```
Comment (jensck) [inline .idea/gradle.xml:15]: `.gitignore` should probably be updated to include all these other automatically re-generated Android Studio files.

File: cloudsdk/src/main/java/io/particle/android/sdk/cloud/models/SignUpInfo.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+/**
+ * Required and optional user information used in sign up process.
+ */
+@ParametersAreNonnullByDefault
+public class SignUpInfo implements Parcelable {
+    private String username, password;
```
Comment (jensck) [inline cloudsdk/src/main/java/io/particle/android/sdk/cloud/models/SignUpInfo.java:16]: Inlining multiple field declarations like this makes them easy to miss when skimming code, IMO.

I'm not going to block the PR approval on it, but I don't think it's worth the tradeoff.

(Same goes for the annotations below)

File: cloudsdk/src/main/java/io/particle/android/sdk/cloud/models/AccountInfo.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    public int describeContents() {
+        return 0;
+    }
+
+    @Override
+    public void writeToParcel(Parcel dest, int flags) {
+        dest.writeString(this.firstName);
+        dest.writeString(this.lastName);
+        dest.writeString(this.companyName);
+        dest.writeByte(this.businessAccount ? (byte) 1 : (byte) 0);
```
Comment (jensck) [inline cloudsdk/src/main/java/io/particle/android/sdk/cloud/models/AccountInfo.java:75]: Not a PR approval blocker, but `Parcelables.writeBoolean()` was written for exactly this. :)

**Review by jensck — APPROVED** [review]
There are a couple things I'd recommend changing, but nothing that I'd block approval on.

@CityVibes you can merge whenever you're ready.

**Pull request merged**